### PR TITLE
DbId connections file

### PR DESCRIPTION
**Pull request by rlcee**
To connect to the database for reading, you need url's.  To connect for writing, you need psql node and port.  These parameters were hardcoded, but the ongoing switch of database platforms from sl6 to sl7 shows these should be taken from a source which can be updated for all code versions and running jobs.  To achieve this, the connections information is moved to cvmfs DataFiles/Database/connections.txt, where it can be changed in approximately real time for all users.

### CODE REVIEW

File: DbService/src/DbIdList.cc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	bool fail = host.empty() || port.empty() || url.empty() || urlNoCache.empty();
+	if ( fail ) {
+	  throw cet::exception("DBIDFILE_DATABASE_NOT_LOADED")
+	    << "DbIdList::DbIdList could not load database " << name 
+	    << " in connections file" << std::endl;
+	}
+	_ids.emplace_back(name,host,port,url,urlNoCache);
+	name=host=port=url=urlNoCache="";
+      } // if database
+      if(words.size()>=2) {
```
Comment (gaponenko) [inline DbService/src/DbIdList.cc:46]: suggest to require words.size()==2 and throw on words.size()>2

**Review by gaponenko — APPROVED** [review]
I have a minor comment on the code (below), but overall the changes look straightforward  and I am approving the merge.   From what I heard in the meeting today I expected a more focused patch, but it is good to see create.sql added to the repository.
Thank you,
Andrei

File: DbService/src/DbIdList.cc (hunk 1 above)
Comment (rlcee) [reply DbService/src/DbIdList.cc]: this, and more protections added

**Comment by kutschke** [comment]
Ray: do you want to do anything else before I merge?

From: Ray Culbertson ***@***.***>
Reply-To: Mu2e/Offline ***@***.***>
Date: Wednesday, March 24, 2021 at 6:35 PM
To: Mu2e/Offline ***@***.***>
Cc: Rob Kutschke ***@***.***>, Assign ***@***.***>
Subject: Re: [Mu2e/Offline] DbId connections file (#390)


@rlcee commented on this pull request.

________________________________

In DbService/src/DbIdList.cc<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Offline_pull_390-23discussion-5Fr600942584&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=xpP3QI8dQ-DGoxuVH0YOkTJfZg35yLzIkPHirZPfHLA&s=CEprirnMVrGgDjWLvB4dbs32-EkoxAETWtbL4o6YnTQ&e=>:

> +      boost::trim(line);

+      boost::split(words,line, boost::is_any_of(" \t"), boost::token_compress_on);

+

+      // if eof, or starting a new database stanza with a previous database name filled

+      if(  in.eof() || ( words.size()>0 && words[0] == "database" && !name.empty()) ) {

+ // have read a stanza, check and store it

+ bool fail = host.empty() || port.empty() || url.empty() || urlNoCache.empty();

+ if ( fail ) {

+   throw cet::exception("DBIDFILE_DATABASE_NOT_LOADED")

+     << "DbIdList::DbIdList could not load database " << name

+     << " in connections file" << std::endl;

+ }

+ _ids.emplace_back(name,host,port,url,urlNoCache);

+ name=host=port=url=urlNoCache="";

+      } // if database

+      if(words.size()>=2) {

this, and more protections added

**Comment by rlcee** [comment]
Rob, please proceed

**Comment by kutschke** [comment]
@FNALbuild run build tests

**Pull request merged**